### DISCUSSION TITLE
从KernelSU迁移到APatch，有办法方便地迁移模块而非逐个重新手动安装吗？

### DISCUSSION THREAD

**Discussion by txmu** [bmax121/APatch#354] [Q&A]
如题。KSU并没有类似mm管理器之类的模块用来方便地备份模块。

**Comment by bmax121** [comment]
没有

**Comment by 212721** [comment]
我也在找

**Comment by tony403** [comment]
可以先区分“模块文件迁移”和“模块兼容性迁移”：APatch 的 APM 机制源自 KernelSU，但官方文档明确提醒文件位置、环境变量和 SELinux 支持存在差异，因此不建议把 /data/adb/ksu 下的状态目录整体复制过去。

较安全的做法是：

1. 在 KernelSU 中逐个确认模块来源和版本，优先下载原始模块 ZIP，并保留当前模块列表作为回滚记录。
2. 在 APatch 管理器中逐个安装这些 ZIP，重启后一次只启用/验证一个模块；APatch 官方文档说明模块 ZIP 应通过 APatch 管理器安装，不要在 Recovery 中安装。
3. 重点检查依赖 KernelSU 专属路径（如 /data/adb/ksu/bin/busybox）或 KernelSU 专属 API 的脚本；APatch 的 BusyBox 路径是 /data/adb/ap/bin/busybox，脚本应使用 MODDIR=${0%/*}，不要硬编码旧路径。
4. 依赖 Zygisk 的模块不能直接视为兼容：APatch 没有内置 Zygisk 支持，需要按模块作者提供的 APatch/ZygiskNext 方案单独确认。

如果模块没有公开 ZIP、包含设备专属脚本，或会修改启动链，建议先在可回滚的备份/测试环境验证，不要批量删除 KernelSU 的原模块。官方文档： https://github.com/bmax121/APatch/blob/main/docs/cn/ap_module.md
